packages/webview_flutter/webview_flutter_android/test/v4/android_webview_controller_test.dart

```diff
@@ -0,0 +1,654 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:webview_flutter_android/src/android_webview.dart'
+    as android_webview;
+import 'package:webview_flutter_android/src/v4/src/android_navigation_delegate.dart';
+import 'package:webview_flutter_android/src/v4/src/android_proxy.dart';
+import 'package:webview_flutter_android/src/v4/src/android_webview_controller.dart';
+import 'package:webview_flutter_platform_interface/v4/src/webview_platform.dart';
+
+import 'android_webview_controller_test.mocks.dart';
+
+@GenerateNiceMocks(<MockSpec<Object>>[
+  MockSpec<AndroidNavigationDelegate>(),
```

Comment:
Out of curiosity, why do you use `@GenerateNiceMocks` instead of `@GenerateMocks`?

Reply:
Initially I ran into an issues where I received an exception because I didn't stub a property that I was not explicitly testing but was used by the code under test. I was a bit confused by the exception and only later realised the problem.

During my research I stumbled on the `@GenerateNiceMocks` which solved the problem and made me realise the real problem. I did fix it and stubbed the property but reading the [documentation](https://pub.dev/packages/mockito#nice-mocks-vs-classic-mocks) it mentions the `@GenerateNiceMocks` is the recommended API which is why I switched.